Chameleon Outfit Change

Vimas21 opened the pull request:
:cl: Vimas21
rscadd: The chameleon jumpsuit now has an option to change every equipped piece to match a torch position in one go.
/:cl:

SammasC123 committed:
Preliminary Code

Ported directly from tg. Needs to be adapted to BayCode.
initialize_outfits()

This proc will, if not yet initialized, will return a list of all outfits from the outfit heirarchy.
Chameleon_Under Outfit Choice

The Chameleon jumpsuit now has the verb Change Chameleon Outfit, which allows selection between all crew outfits. This currently does nothing.
End Of Night Push

Menu in, now leads to outfits. For some reason it's leading to null though. Gotta figure that out.

More Chameleon Changes

I don't know how to write these ah
Sprite Changing

Sprites Change! Just, uh- only the suit. For some reason.
Almost Done

Sprites change- they just need to update.
End of Night Push

I'm so fucking close >:(
Under- Chameleon Outfit Selection

That's right, it's done bitches! The jumpsuit in the chameleon suit now has a button- Chameleon Outfit Change. This gives a list of all torch jobs, and allows you to select one. All equipped chameleon objects that have a valid equivalent in the given job change in one go!
Bug Quashing

Going through and fixing each bug/eccentricity commented on in git
More Tweaks

See suggestions on git.

PsiOmegaDelta commented on code/modules/clothing/chameleon.dm:
```diff
@@ -65,6 +83,37 @@
+		return FALSE
+	var/list/outfits = initialize_outfits()
+	var/selected_key = input(usr, "Choose an Outfit", "Chameleon Outfit") as null|anything in outfits
+	if (!selected_key)
```
Need to recheck that the user is still not incapable after the `input()`

SierraKomodo replied:
This still needs to be addressed

PsiOmegaDelta commented on code/modules/clothing/chameleon.dm:
```diff
@@ -65,6 +83,37 @@
+	if(istype(user.wear_mask, /obj/item/clothing/mask/chameleon) && initial(selected.mask))
+		user.wear_mask.disguise(initial(selected.mask),usr)
+	if(istype(user.l_ear, /obj/item/device/radio/headset/chameleon) && initial(selected.l_ear))
+		user.l_ear.disguise(initial(selected.l_ear),usr)
```
I'm going to ponder an alternative to this block of copy-paste, because it's simply not acceptable.

Vimas21 replied:
Is that you offering to fix it, or is it something I should be looking into? 😎

PsiOmegaDelta commented on code/modules/clothing/chameleon.dm:
```diff
@@ -65,6 +83,37 @@
+	if(istype(user.w_uniform, /obj/item/clothing/under/chameleon) && initial(selected.uniform))
+		disguise(initial(selected.uniform),usr)
+	if(istype(user.head, /obj/item/clothing/head/chameleon) && initial(selected.head))
+		user.head.disguise(initial(selected.head),usr)
```
Also, mixed use of `usr` and `user`

SammasC123 committed:
Psi Remarks

Added a doublecheck for incapacitation, and straightened out the use of user vs usr.

SierraKomodo commented:
Closing for #29027
